packages/server/src/cds/routes.ts

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: Copyright Orangebot, Inc. and Medplum contributors
+// SPDX-License-Identifier: Apache-2.0
+import { allOk, badRequest, ContentType, getStatus, Operator } from '@medplum/core';
+import type { Bot } from '@medplum/fhirtypes';
+import type { Request, Response } from 'express';
+import { Router } from 'express';
+import { executeBot } from '../bots/execute';
+import { getBotDefaultHeaders, getResponseBodyFromResult } from '../bots/utils';
+import { getAuthenticatedContext } from '../context';
+import { authenticateRequest } from '../oauth/middleware';
+
+// CDS Hooks
+// https://cds-hooks.hl7.org/
+
+export const cdsRouter = Router().use(authenticateRequest);
+
+// Discovery: https://cds-hooks.hl7.org/#discovery
+cdsRouter.get('/', async (_req: Request, res: Response) => {
+  const { repo } = getAuthenticatedContext();
+  repo.setExtendedMode(true);
+
+  // The CDS Hooks spec does not define pagination for discovery
+  // Most servers will have under 10 CDS services, so we set a high max count.
+  const maxCount = 100;
+
+  const bots = await repo.searchResources<Bot>({
+    resourceType: 'Bot',
+    filters: [{ code: 'cds-hook', operator: Operator.MISSING, value: 'false' }],
+    count: maxCount,
+  });
+
+  // All bots will have a cdsService defined due to the search filter
+  const response = {
+    services: bots.map((bot) => ({
+      id: bot.id,
+      hook: bot.cdsService?.hook,
+      title: bot.cdsService?.title,
+      description: bot.cdsService?.description,
+      usageRequirements: bot.cdsService?.usageRequirements,
+      prefetch: bot.cdsService?.prefetch
+        ? Object.fromEntries(bot.cdsService.prefetch.map(({ key, query }) => [key, query]))
+        : undefined,
+    })),
+  };
+
+  res.status(200).json(response);
+});
+
+// Calling a CDS Service: https://cds-hooks.hl7.org/#calling-a-cds-service
+cdsRouter.post('/:id', async (req: Request, res: Response) => {
+  const { id } = req.params as { id: string };
+
+  const ctx = getAuthenticatedContext();
+  ctx.repo.setExtendedMode(true); // Need extended mode for meta.project
```

Comment:
Looking around a bit, it appears that other places where we are performing `executeBot` we fetch the resource by using the system repo, which has extendedMode set.

Does it make sense to follow that same pattern here? Or to adjust other places to match this behavior of using a scoped repo with the extended mode flag set? (Not a blocker, just trying to understand the direction the code is moving in so I can match it going forward)

Are there things like middleware that might be using this context repo that we don't want to also receive extended mode behavior? Should we do something like `ctx.repo.clone()` before mutating to isolate this change?